src/sqlfluff/rules/L050.py

```diff
@@ -91,34 +71,29 @@ def _eval(self, context: RuleContext) -> Optional[LintResult]:
         if len(context.raw_stack) == 0:
             return None
 
-        # If the current segment is either comment or code and all
-        # previous segments are forms of whitespace then we can
-        # remove these earlier segments.
-        # Given the tree stucture, we make sure we are at the
-        # first leaf to avoid repeated detection.
-        whitespace_set = {"newline", "whitespace", "Indent", "Dedent"}
+        segment = context.surrogates.segment
+        raw_stack = context.surrogates.raw_stack
+        whitespace_types = {"newline", "whitespace", "indent", "dedent"}
+        # Non-whitespace segment.
         if (
             # Non-whitespace segment.
-            (context.segment.name not in whitespace_set)
+            not segment.all(*whitespace_types)
             # We want first Non-whitespace segment so
             # all preceding segments must be whitespace
             # and at least one is not meta.
-            and all(segment.name in whitespace_set for segment in context.raw_stack)
-            and not all(segment.is_meta for segment in context.raw_stack)
+            and raw_stack.all(*whitespace_types)
+            and not raw_stack.all(lambda s: s.is_meta)
             # Found leaf of parse tree.
-            and (not context.segment.is_expandable)
-        ):
+            and not segment.all(lambda s: s.is_expandable)
             # It is possible that a template segment (e.g. {{ config(materialized='view') }})
             # renders to an empty string and as such is omitted from the parsed tree.
             # We therefore should flag if a templated raw slice intersects with the
             # source slices in the raw stack and skip this rule to avoid risking
             # collisions with template objects.
-            if self._potential_template_collision(context):
-                return None
-
+            and not raw_stack.raw_slices.any("templated")
```

Comment:
Interesting that this doesn't seem to be triggering the dbt test case with the leading invisible `{{ config(materialized="view") }}` that was causing me issues when I first made this rule. But if it works then that's great haha 🤷

Reply:
Oh nevermind, read it as any templated segments in the raw stack rather than raw slices as you'd added. So it encapsulates the same logic I originally had. Nice!

Reply:
Yes, exactly the same, just... shorter code. 🎉